aten/src/ATen/core/dispatch/Dispatcher.cpp

```diff
@@ -86,69 +88,123 @@ OperatorHandle Dispatcher::findOrRegisterSchema_(FunctionSchema&& schema) {
   return handle;
 }
 
-std::pair<RegistrationHandleRAII, OperatorHandle> Dispatcher::registerSchema(FunctionSchema schema) {
+// Postcondition: caller is responsible for disposing of registration when they
+// are done
+OperatorHandle Dispatcher::findOrRegisterName_(const OperatorName& op_name) {
+  const auto found = findSchema(op_name);
+  if (found != c10::nullopt) {
+    return *found;
+  }
+
+  operators_.emplace_back(OperatorName(op_name));
```

Comment:
nit: `push_back(op_name)` is shorter than `emplace_back(OperatorName(op_name))` and has one less constructor call (the emplace_back version does call a move after you explicitly copy it while push_back only copies once and never moves)

Reply:
Actually, this is not valid here, because the constructor is explicitly written to only accept rvalue references.

```
      explicit OperatorDef(FunctionSchema&& schema)
      : op(std::move(schema)) {}
    
      explicit OperatorDef(OperatorName&& op_name)
      : op(std::move(op_name)) {}
    
```

I'm not too sure why it was originally written this way; I guess they wanted us to not do a copy accidentally. A copy is needed here so I have to turn the operator name into an rvalue and then move it.

Reply:
I missed that `operator_` takes `OperatorDef` and not `OperatorName`. Never mind my comment.